sdk/core/core-lro/review/core-lro.api.md

```diff
@@ -10,53 +10,45 @@ import { AbortSignalLike } from '@azure/abort-controller';
 export type CancelOnProgress = () => void;
 
 // @public
-export function createHttpPoller<TResult, TState extends OperationState<TResult>>(lro: LongRunningOperation, options?: CreateHttpPollerOptions<TResult, TState>): Promise<SimplePollerLike<TState, TResult>>;
+export function createHttpPoller<TResult, TState extends OperationState<TResult>>(lro: LongRunningOperation, options?: CreateHttpPollerOptions<TResult, TState>): PollerLike<TState, TResult>;
 
 // @public
 export interface CreateHttpPollerOptions<TResult, TState> {
     intervalInMs?: number;
-    processResult?: (result: unknown, state: TState) => TResult;
+    processResult?: (result: unknown, state: TState) => TResult | Promise<TResult>;
     resolveOnUnsuccessful?: boolean;
-    resourceLocationConfig?: LroResourceLocationConfig;
+    resourceLocationConfig?: ResourceLocationConfig;
     restoreFrom?: string;
-    updateState?: (state: TState, response: LroResponse) => void;
+    updateState?: (state: TState, response: OperationResponse) => void;
     withOperationLocation?: (operationLocation: string) => void;
 }
 
+// @public
+export function deserializeState<TState>(serializedState: string): RestorableOperationState<TState>;
+
 // @public
 export interface LongRunningOperation<T = unknown> {
-    requestMethod?: string;
-    requestPath?: string;
-    sendInitialRequest: () => Promise<LroResponse<unknown>>;
+    sendInitialRequest: () => Promise<OperationResponse<unknown>>;
```

Comment:
no longer need `requestMethod?: string; requestPath?: string;` ?

Reply:
yeah, the method and path info could be devired from initial prepared request and no need to be provided as input values.

another consideration is that we changed the design of restore poller from operation level to client level. so we save the method and path info as part of poller serialization detail.